test/e2e/issue279_test.go

```diff
@@ -0,0 +1,301 @@
+		const name = "gitopsconfig-gitops-issue279"
+		cronJob, err := GetCronJob(namespace, name, framework.Global.KubeClient)
+		switch {
+		case apierrors.IsNotFound(err) || cronJob == nil:
+			t.Logf("CronJob %s successfully deleted", name)
+			return true, nil
+		case err != nil:
+			return false, err
+		case cronJob != nil:
+			t.Logf("Waiting for cronJob %s to be deleted", cronJob.Name)
+			return false, nil
+		default:
+			t.Logf("Waiting for cronJob %s to be deleted", name)
+			return false, nil
+		}
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+}
+
+// TestIssue279CRWithPeriodicDeletion verifies that after removing GitOpsConfig
+// with Periodic trigger, CronJob also gets deleted
+func TestIssue279CRWithPeriodicDeletion(t *testing.T) {
+	ctx := framework.NewTestCtx(t)
+	defer ctx.Cleanup()
+
+	namespace, err := ctx.GetNamespace()
+	if err != nil {
+		t.Fatalf("could not get namespace: %v", err)
+	}
+	if err = SetupRbacInNamespace(namespace); err != nil {
+		t.Error(err)
+	}
+
+	defer DumpJobsLogsOnError(t, framework.Global, namespace)
+	err = framework.AddToFrameworkScheme(apis.AddToScheme, &gitopsv1alpha1.GitOpsConfigList{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	eunomiaURI, found := os.LookupEnv("EUNOMIA_URI")
+	if !found {
+		eunomiaURI = "https://github.com/kohlstechnology/eunomia"
+	}
+	eunomiaRef, found := os.LookupEnv("EUNOMIA_REF")
+	if !found {
+		eunomiaRef = "master"
+	}
+
+	// Step 1: create a CR with a Periodic trigger
+
+	gitops := &gitopsv1alpha1.GitOpsConfig{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "GitOpsConfig",
+			APIVersion: "eunomia.kohls.io/v1alpha1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "gitops-issue279",
+			Namespace: namespace,
+			Finalizers: []string{
+				"gitopsconfig.eunomia.kohls.io/finalizer",
+			},
+		},
+		Spec: gitopsv1alpha1.GitOpsConfigSpec{
+			TemplateSource: gitopsv1alpha1.GitConfig{
+				URI:        eunomiaURI,
+				Ref:        eunomiaRef,
+				ContextDir: "test/e2e/testdata/empty-directory",
+			},
+			ParameterSource: gitopsv1alpha1.GitConfig{
+				URI:        eunomiaURI,
+				Ref:        eunomiaRef,
+				ContextDir: "test/e2e/testdata/empty-yaml",
+			},
+			Triggers: []gitopsv1alpha1.GitOpsTrigger{
+				{
+					Type: "Periodic",
+					Cron: "*/1 * * * *",
+				},
+			},
+			TemplateProcessorImage: "quay.io/kohlstechnology/eunomia-base:dev",
+			ResourceHandlingMode:   "Apply",
+			ResourceDeletionMode:   "Delete",
+			ServiceAccountRef:      "eunomia-operator",
+		},
+	}
+	gitops.Annotations = map[string]string{"gitopsconfig.eunomia.kohls.io/initialized": "true"}
+
+	err = framework.Global.Client.Create(context.TODO(), gitops, &framework.CleanupOptions{TestContext: ctx, Timeout: timeout, RetryInterval: retryInterval})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Step 2: Make sure that CronJob has been created
+
+	err = wait.Poll(retryInterval, timeout, func() (done bool, err error) {
+		const name = "gitopsconfig-gitops-issue279"
```

Comment:
similar as above, `-` suffix?

Reply:
For some reason I don't know the cronJob in the second test has the exact name "gitopsconfig-gitops-issue279". If I remove the suffix, test fails.